### PR TITLE
Control access to recordings and timers

### PR DESCRIPTION
**Pull request by emveepee**
Provide options to access Live TV only (no recording or timer access) or to disable timers only.

Add new timer type single channel all episodes (filtered for new and duplicate  at the backend)  and Daily Timeslot recording.

### CODE REVIEW

**Review by phunkyfish — APPROVED** [review]

File: pvr.nextpvr/addon.xml.in [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <addon
   id="pvr.nextpvr"
-  version="21.0.3"
+  version="21.0.4"
```
Comment (ksooo) [inline pvr.nextpvr/addon.xml.in]: Usually we increase the minor version number when we introduce new features, not the micro version.

Comment (emveepee) [reply pvr.nextpvr/addon.xml.in]: Fair enough, I considered it a minor change, but I have changed it.  All the .0 releases for new changes could look funny.

**Comment by emveepee** [comment]
@phunkyfish could you please re-review.  I revamped the original commit to one setting rather than two.  I found a few hardcoded English string in instance-settings.xml that I was lax on updating.  I also fixed an outstanding issue with missing timer types that had not been added to Kodi.

File: pvr.nextpvr/addon.xml.in (hunk 1 above)
Comment (ksooo) [reply pvr.nextpvr/addon.xml.in]: New feature => increase minor, set micro to zero
Bug fix only => increase micro 

This is how many software projects handle it. So do we.

**Comment by ksooo** [comment]
Don’t want to block this PR. only out of curiosity: does nextpvr backend not have a multi user concept? Different users can log in with different credentials, having their own set of permissions/config? For me, this would be a clean way to implement what the user requested in the forum. Nothing that should be done by the addon, imo, if the backend offers multi user support.

File: pvr.nextpvr/changelog.txt [hunk 2]
```diff
@@ -1,5 +1,8 @@
 v21.0.4
 - Allow control of recording and timers access
+- Strings.po was missing some hardcoded settings values
```
Comment (phunkyfish) [inline pvr.nextpvr/changelog.txt]: The changelog is for users really. So this is not infomation they really need. It's no harm to have it but not required.

File: src/pvrclient-nextpvr.cpp [hunk 3]
```diff
@@ -1020,11 +1028,11 @@ PVR_ERROR cPVRClientNextPVR::GetCapabilities(kodi::addon::PVRCapabilities& capab
   kodi::Log(ADDON_LOG_DEBUG, "->GetCapabilities()");
 
   capabilities.SetSupportsEPG(true);
-  capabilities.SetSupportsRecordings(true);
-  capabilities.SetSupportsRecordingsDelete(true);
+  capabilities.SetSupportsRecordings(m_settings->m_accessLevel > 0);
```
Comment (phunkyfish) [inline src/pvrclient-nextpvr.cpp]: Plese avoid the use of literal numbers. Define a value to make the code more readable.

File: src/pvrclient-nextpvr.cpp [hunk 4]
```diff
@@ -1020,11 +1028,11 @@ PVR_ERROR cPVRClientNextPVR::GetCapabilities(kodi::addon::PVRCapabilities& capab
...
+  capabilities.SetSupportsRecordingsDelete(m_settings->m_accessLevel > 1);
```
Comment (phunkyfish) [inline src/pvrclient-nextpvr.cpp]: Plese avoid the use of literal numbers. Define a value to make the code more readable.

File: src/pvrclient-nextpvr.cpp [hunk 5]
```diff
@@ -1020,11 +1028,11 @@ PVR_ERROR cPVRClientNextPVR::GetCapabilities(kodi::addon::PVRCapabilities& capab
...
   capabilities.SetSupportsRecordingsUndelete(false);
   capabilities.SetSupportsRecordingSize(m_settings->m_showRecordingSize);
-  capabilities.SetSupportsTimers(true);
+  capabilities.SetSupportsTimers(m_settings->m_accessLevel == 3);
```
Comment (phunkyfish) [inline src/pvrclient-nextpvr.cpp]: Plese avoid the use of literal numbers. Define a value to make the code more readable.

**Review by phunkyfish — CHANGES_REQUESTED** [review]

**Comment by emveepee** [comment]
> Don’t want to block this PR. only out of curiosity: does nextpvr backend not have a multi user concept? Different users can log in with different credentials, having their own set of permissions/config? For me, this would be a clean way to implement what the user requested in the forum. Nothing that should be done by the addon, imo, if the backend offers multi user support.

No the backend doesn't support multi-user.  If I had access to the backend source this is probably the first thing that I would change.  As an example having one resume point for a recording can cause conflicts, which is why I added an option to store the resume point locally.   That all being said over the almost 20 years I have been involved with the project there have been very few calls to add multi-user support.  I think users like the simplicity of installing NextPVR and most wouldn't be bothered to use the feature anyway.

File: src/pvrclient-nextpvr.cpp (hunk 3 above)
Comment (emveepee) [reply src/pvrclient-nextpvr.cpp]: Reworked these values as flags.

**Review by phunkyfish — APPROVED** [review]

**Pull request merged**